data.tf

```diff
@@ -7,8 +7,13 @@ data "cloudfoundry_user" "user" {
   org_id = data.cloudfoundry_org.org.id
 }
 
+data "hsdp_config" "cf" {
+  region = var.cf_region
+  service = "cf"
+}
+
 data "cloudfoundry_domain" "app_domain" {
-  name = var.cf_app_domain
+  name = data.hsdp_config.cf.domain
```

Comment:
What happens here now if I supply `var.cf_app_domain` looks like it would be ignored?

Reply:
Yes, but as you suggested, it will be removed completely and a new TAG will be created to deal with backwards compatibility